Client/src/main/java/gui/panels/boards/BoardsFactory.java

```diff
@@ -0,0 +1,51 @@
+package gui.panels.boards;
+
+import gui.services.BoardPanelType;
+import models.GameState;
+import gui.services.Publisher;
+import org.springframework.beans.factory.annotation.Autowired;
+import services.shared.BoardsMessageService;
+
+import javax.annotation.PostConstruct;
+import javax.swing.*;
+import java.awt.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BoardsFactory {
+
+    @Autowired
+    private BoardsMessageService boardsMessageService;
+
+    @Autowired
+    private UserBoard userBoard;
+
+    @Autowired
+    private RivalBoard rivalBoard;
+
+    @Autowired
+    private Publisher boardPublisher;
+
+
+    private Map<BoardPanelType, JPanel> boards = new HashMap<>();
+
+    @PostConstruct
+    public void boardFactoryInitialize() {
+        initBoardsMap();
+    }
+
+    public JPanel getBoardPanel(BoardPanelType boardPanelType) {
+        return boards.getOrDefault(boardPanelType, new JPanel());
+    }
+
+    private void initBoardsMap() {
+        boards.put(BoardPanelType.Playing, createPlayingBoard());
+    }
+
+    private JPanel createPlayingBoard() {
```

Comment:
I wonder about the name here. I'd say client window might be a better fit? After all, playing board does not consist of user board AND rival board AND panel with game state...